### ISSUE TITLE
Underline error span instead of just caret (if possible) when using --pretty

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#7454]
It would be great instead of:
```
mypy/semanal.py:1052: error: Argument 1 to "analyze_base_classes" of "SemanticAnalyzer"
has incompatible type "List[Node]"; expected "List[Expression]"  [arg-type]
            result = self.analyze_base_classes(bases)
                                               ^
```
show something like
```
mypy/semanal.py:1052: error: Argument 1 to "analyze_base_classes" of "SemanticAnalyzer"
has incompatible type "List[Node]"; expected "List[Expression]"  [arg-type]
            result = self.analyze_base_classes(bases)
                                               ^~~~~
```
We could use some heuristics and/or Python 3.8 support for `end_col_offset`:
```python
    if isinstance(context, NameExpr):
        end_column = context.column + len(context.name)
    elif options.python_version >= (3, 8):
        end_column = context.end_column
    ...
```

This is a follow-up for https://github.com/python/mypy/pull/7440

**Comment by gvanrossum** [comment]
I would only do this if we have definitive end-col info (i.e. 3.8+).

**Comment by BenC14** [comment]
Would also love to see this, so my IDE can actually underline the nodes completely.

**Comment by hauntsaninja** [comment]
Fixed by #13148
